### PR TITLE
charmpp: add archs including Cray shasta with ARM

### PR DESCRIPTION
**Pull request by RMeli**
* Add patch to build `arch/ofi-crayshasta` on ARM
* Add `cray-pmi` as PMI provider
* Add `cray-mpich` dependency when PMI provider is `cray-pmi` (similar to `openmpi` when provider is `pmix`)
* Add explicit dependency on `libfabric` when `backend=ofi`
* Added a bunch of versions
* Added `-smp` suffix when `+smp`, for NAMD

### CODE REVIEW

**Comment by RMeli** [comment]
@spackbot fix style

**Review by bernhardkaindl — APPROVED** [review]
@matthiasdiener @RMeli: 8 Days since this was opened, ready to merge?

@RMeli It looks like a lot of improvements, maybe add a summary to the PR description for people to read?

**Comment by RMeli** [comment]
@bernhardkaindl I added a list of changes, as requested. Before merging I'd really like feedback from @matthiasdiener (there is no rush from my side).

**Comment by RMeli** [comment]
@matthiasdiener, is there any chance you could have a look at this PR? Thank you.

**Comment by RMeli** [comment]
@jcphill would you mind having a quick look at this PR? Thank you.

**Review by matthiasdiener — APPROVED** [review]
LGM, thanks! Sorry for the delay.

**Review by alalazo — APPROVED** [review]
Maintainer approved, recipe LGTM

**Comment by RMeli** [comment]
Thanks @matthiasdiener!

**Pull request merged**
